src/main/java/seedu/address/model/event/Event.java

```diff
@@ -0,0 +1,124 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+import java.util.Objects;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.UniquePersonList;
+
+/**
+ * Represents an Event in the event book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Event {
+
+    private final EventName eventName;
+    private final UniquePersonList persons;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Event(EventName eventName) {
+        requireAllNonNull(eventName);
+        this.eventName = eventName;
+        persons = new UniquePersonList();
+    }
+
+    public EventName getEventName() {
+        return eventName;
+    }
+
+    //// person-level operations
+
+    /**
+     * Replaces the contents of the person list with {@code persons}.
+     * {@code persons} must not contain duplicate persons.
+     */
+    public void setPersons(List<Person> persons) {
+        this.persons.setPersons(persons);
+    }
+
+    /**
+     * Returns true if a person with the same identity as {@code person} exists in the address book.
+     */
+    public boolean hasPerson(Person person) {
+        requireNonNull(person);
+        return persons.contains(person);
+    }
+
+    /**
+     * Adds a person to the event.
+     * The person must not already exist in the event.
+     */
+    public void addPerson(Person p) {
```

Comment:
Using p is okay here but i do feel like using `person` makes the code appear more consistent.